shared/cache.ts

```diff
@@ -0,0 +1,61 @@
+import { DateTime } from "luxon";
+
+type Cache<DataType> = {
+  value?: DataType;
+  populatedAt?: string;
```

Comment:
Not married to this but thoughts on making this a DateTime type?
I don't see the benefit of storing it as a string and then having to constantly convert it to a datetime object.

Reply:
I think it's best if we store it as a string. There's a couple of benefits to that (although typescript might make them unnecessary?):
- It's clear that _every_ datetime property needs to be converted first, so it's consistent everywhere
- We know we can serialize it, if we ever need to send it across the wire (note that if it every makes it to page props it'll need to be `json` serializable

Neither of those advantages _really_ apply in this case, but I think we should follow a strategy that's consistent everywhere, so we're not guessing why a particular part is different.

I also don't think there's too much overhead in converting it - since anywhere we convert it we need to use other luxon utilities (like formatting or getting time differential) anyway.

Reply:
yea lets do the audit in a different PR. 

I agree with some of you above points but not really fussed so happy to go with it :)